src/rp/index.ts

```diff
@@ -0,0 +1,143 @@
+import { AuthRequestDecodeError, IoWalletError } from "../utils/errors";
+import {
+  decode as decodeJwt,
+  decodeBase64,
+  sha256ToBase64,
+  SignJWT,
+} from "@pagopa/io-react-native-jwt";
+import { RequestObject, RpEntityConfiguration } from "./types";
+
+import uuid from "react-native-uuid";
+import type { JWK } from "@pagopa/io-react-native-jwt/lib/typescript/types";
+
+export class RelyingPartySolution {
+  relyingPartyBaseUrl: string;
+  walletInstanceAttestation: string;
+  appFetch: GlobalFetch;
+
+  constructor(
+    relyingPartyBaseUrl: string,
+    walletInstanceAttestation: string,
+    appFetch: GlobalFetch = { fetch }
+  ) {
+    this.relyingPartyBaseUrl = relyingPartyBaseUrl;
+    this.walletInstanceAttestation = walletInstanceAttestation;
+    this.appFetch = appFetch;
+  }
+
+  /**
+   * Decode a QR code content to an authentication request url.
+   * @function
+   * @param qrcode QR code content
+   *
+   * @returns The authentication request url
+   *
+   */
+  decodeAuthRequestQR(qrcode: string): string {
+    let decoded = decodeBase64(qrcode);
+    let decodedUrl = new URL(decoded);
+    let requestUri = decodedUrl.searchParams.get("request_uri");
```

Comment:
Why `let`? (I also wonder why `eslint` doesn't complain, as we should have a rule to prefer `const`)

Reply:
fixed